daffodil-test/src/test/resources/org/apache/daffodil/layers/ipv4.dfdl.xsd

```diff
@@ -0,0 +1,261 @@
+          A dfdl:outputValueCalc puts this newly computed value from the variable into the checksum element.
+          Any prior value of the checksum element is discarded/overwritten with the new value.
+
+          -->
+          <dfdl:newVariableInstance ref="chksum:ipV4ChecksumPart1Value"/>
+          <dfdl:newVariableInstance ref="chksum:ipV4ChecksumValue"/>
+        </xs:appinfo>
+      </xs:annotation>
+
+      <xs:sequence dfdl:ref="chksum:ipv4ChecksumLayerPart1">
+        <xs:sequence>
+        <!--
+        Within this sequence, the data is all included in Part 1 of the
+        ipV4 checksum calculation
+
+        On parsing, the checksum algorithm computes the partial checksum for
+        this first part of the data, placing the value into the
+        variable chksum:checksumPart1Value, which must have a new variable
+        instance in scope.
+
+        This checksum calculation knows exactly how big this data must be
+        and if the representation turned out to be a different length than this, it
+        will be a runtime SDE.
+
+        Everything from here to the end of this sequence is unmodified from
+        the original IPV4 schema it is based on.
+        -->
+        <xs:element name="Version" type="tns:bit" dfdl:length="4"/>
+        <xs:element name="IHL" type="tns:bit" dfdl:length="4"/>
+        <xs:element name="DSCP" type="tns:bit" dfdl:length="6"/>
+        <xs:element name="ECN" type="tns:bit" dfdl:length="2"/>
+        <xs:element name="Length" type="tns:bit" dfdl:length="16"/>
+        <xs:element name="Identification" type="tns:bit" dfdl:length="16"/>
+        <xs:element name="Flags" type="tns:bit" dfdl:length="3"/>
+        <xs:element name="FragmentOffset" type="tns:bit" dfdl:length="13"/>
+        <xs:element name="TTL" type="tns:bit" dfdl:length="8"/>
+        <xs:element name="Protocol" type="tns:bit" dfdl:length="8"/>
+        </xs:sequence>
+      </xs:sequence>
+      <!--
+      Now the checksum element itself.
+
+      When unparsing: This is computed via a forward reference to the
+      checksum computation result in the
+      checksumValue variable, which is computed by the checksumLayerPart2 transform.
+      -->
+      <xs:element name="Checksum" type="chksum:ipv4ChecksumType"
+                  dfdl:outputValueCalc='{ $chksum:ipV4ChecksumValue }'/>
+      <!--
+      Now the part 2 layer transform is computed. This combined the result from
+      the part 1 layer transform (from the checksumPart1Value variable) with the
+      remaining checksum computed from the sequence content.
+
+      This checksum calculation knows exactly how big this data must be
+      and if the representation turned out to be a different length than this, it
+      will be a runtime SDE.
+
+      The result is placed in the checksumValue variable.
+      -->
+      <xs:sequence dfdl:ref="chksum:ipv4ChecksumLayerPart2">
+      <xs:sequence>
+        <xs:element name="IPSrc" type="tns:hexByte" dfdl:length="4"/>
```

Comment:
enhance so at least one of these does a forward reference to a field later, so as to illustrate suspending during unparse during this part 2 layer transform.